### DISCUSSION TITLE
Error 404 when installing with Apache

### DISCUSSION THREAD

**Discussion by airbornedan** [Kovah/LinkAce#503] [Support]
I'm probably overlooking something stupid, but I can't seem to figure it out, so I thought I'd ask.

I followed the install directions, but I've hit a snag. After unzipping the archive, changing permissions and ownership, and copying the .env file, I navigated to http://servername/linkace/public (where the LinkAce files are located) but am getting a 404 error (at the address http://servername/linkace/public/setup/start). I assume this redirect is supposed to open the configuration page, but there's no /setup/start directory on the server.

Advice?

**Comment by Kovah** [comment]
The web server must have the document root to the linkace/public directory. You must then open servername.com/linkace.

However, the installation in a subdirectory is currently not official supported.

**Comment by RickFr410** [comment]
Hello,
I have the same issue. The Apache's vhost is linked to the public directory. The issue is the same after enabling the htaccess, or put the .htaccess content on the vhost. I don't have a setup directory. Is the path managed by Laravel ?

**Reply by Kovah** [reply]
Yes, all paths are managed by the application. There should not be a setup directory.
If you see any errors, please post the exact error and your configuration.

**Comment by RickFr410** [comment]
the error in the browser is simply a 404 error. This is the Apache's error.log : 

> [Mon Aug 15 08:52:01.947092 2022] [core:info] [pid xxxxxxx:tid 2967802880] [client 192.168.x.x:62581] AH00128: File does not exist: /var/www/html/linkace/public/setup/start
> [Mon Aug 15 08:52:03.116970 2022] [core:info] [pid xxxxxx:tid 2959410176] [client 192.168.x.x:62581] AH00128: File does not exist: /var/www/html/linkace/public/setup/start

The Vhost : 
```
<VirtualHost *:80>

        ServerName xxxxxxxxxxxxxxx
        ServerAdmin web@xxxxxxxxxx

        DocumentRoot /var/www/html/linkace/public

        ErrorLog ${APACHE_LOG_DIR}/error.log
        CustomLog ${APACHE_LOG_DIR}/access.log combined

        <Directory /var/www/html/linkace/public>
                <IfModule mod_rewrite.c>
                    <IfModule mod_negotiation.c>
                        Options -MultiViews -Indexes
                    </IfModule>

                    RewriteEngine On

                    # Handle Authorization Header
                    RewriteCond %{HTTP:Authorization} .
                    RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]

                    # Redirect Trailing Slashes If Not A Folder...
                    RewriteCond %{REQUEST_FILENAME} !-d
                    RewriteCond %{REQUEST_URI} (.+)/$
                    RewriteRule ^ %1 [L,R=301]

                    # Handle Front Controller...
                    RewriteCond %{REQUEST_FILENAME} !-d
                    RewriteCond %{REQUEST_FILENAME} !-f
                    RewriteRule ^ index.php [L]
                </IfModule>
        </Directory>
</VirtualHost>

```

The HTTPS config iqs the same except the SSL config. I tested in HTTP. The .env file is present in the linkace directory and hasn't been modify except the APP_KEY variable generated by the php artisan script.

**Reply by Kovah** [reply]
I guess I can't help with that. Haven't used Apache in like 10 years and can't see any obvious issue.

**Comment by RickFr410** [comment]
Is it possible to enable logs in your script ?

**Comment by RickFr410** [comment]
I enabled the debug log level on Apache. I suspect that it's due to the call of the setup script as a path. I had problems in the past with a joomla on another server and had to use the Apache PHP module. I'll investigate on that.

**Reply by Kovah** [reply]
The Apache PHP module is required to run LinkAce.
